osu.Game/Screens/Edit/Compose/Components/Timeline/TimelineBlueprintContainer.cs

```diff
@@ -61,6 +86,9 @@ protected override void LoadComplete()
 
             placement = beatmap.PlacementObject.GetBoundCopy();
             placement.ValueChanged += placementChanged;
+
+            updateHoverState();
+            FinishTransforms(true);
```

Comment:
What's this doing? Why is the hover state updated in LoadComplete? This drawable shouldn't be in the input hierarchy by this point?

Is this done just because `backgroundBoxIntro` doesn't have `Colour = Color4.Black` like `backgroundBox` does?

Reply:
Just to keep the colour logic in one place, yep.

Not sure what you mean by input hierarchy. This is generally how we handle this kind of state logic elsewhere in the game.

Reply:
As in, there's no way that `IsHovered` can be true, so the only case is `false`. So these two lines and logic flow are avoiding that one line, and I disagree with doing it this way.

Can you point me to another place where this is done? OverlayContainer is an exception because of the initially-shown/initially-hidden thing.

Reply:
I'm very confused. Do you want the gradient in a private field instead? Or defined in two places?

Reply:
Examples of the same kind of pattern:

https://github.com/ppy/osu/blob/1aff263419080160c9bbe078fb26005ae41ef0cb/osu.Game/Screens/Edit/Compose/Components/SelectionBoxDragHandle.cs#L47
https://github.com/ppy/osu/blob/c3757a4660f1b1f1633e9b16af75d2960b343006/osu.Game/Beatmaps/Drawables/UpdateableBeatmapSetCover.cs#L59
https://github.com/ppy/osu/blob/1fd76ea3fb9db1d7e80e92fbd9e9cdb40353683c/osu.Game/Graphics/UserInterface/DrawableOsuMenuItem.cs#L81
https://github.com/ppy/osu/blob/72562070bcea0718bd6ddce220e354d934b6a987/osu.Game/Overlays/Notifications/ProgressNotification.cs#L43

Reply:
Jesus too much effort spent on typing here.

There's supposed to be a gradient, I missed it. This will look wrong when a waveform is visible as it stands.

Reply:
Have refactored to avoid the need for this in the first place.